File: PST/FUEL/src/FUEL/TestSpecs.vb
Imports System
Imports System.Collections.Generic
Imports System.Diagnostics
Imports System.Runtime.CompilerServices

Namespace FUEL
    Public Class TestSpecs
        ' Methods
        Public Sub New()
            Me._MechChecks = New List(Of PrinterMechChecks)
        End Sub


        ' Properties
        Private Property _TestID As String
            <DebuggerNonUserCode> _
            Get
                Return Me.__TestID
            End Get
            <DebuggerNonUserCode> _
            Set(ByVal AutoPropertyValue As String)
                Me.__TestID = AutoPropertyValue
            End Set
        End Property

        Private Property _MechChecks As List(Of PrinterMechChecks)
            <DebuggerNonUserCode> _
            Get
                Return Me.__MechChecks
            End Get
            <DebuggerNonUserCode> _
            Set(ByVal AutoPropertyValue As List(Of PrinterMechChecks))
                Me.__MechChecks = AutoPropertyValue
            End Set
        End Property

        Private Property _PST As PST
            <DebuggerNonUserCode> _
            Get
                Return Me.__PST
            End Get
            <DebuggerNonUserCode> _
            Set(ByVal AutoPropertyValue As PST)
                Me.__PST = AutoPropertyValue
            End Set
        End Property

        Public Property TestID As String
            Get
                Return If((Me.PST.TestID = Nothing), Me._TestID, Me.PST.TestID)
            End Get
            Set(ByVal value As String)
                Me._TestID = value
            End Set
        End Property

        Public Property PST As PST
            Get
                Return Me._PST
            End Get
            Set(ByVal value As PST)
                Me._PST = value
            End Set
        End Property


        ' Fields
        <DebuggerBrowsable(DebuggerBrowsableState.Never), CompilerGenerated> _
        Private __TestID As String
        <DebuggerBrowsable(DebuggerBrowsableState.Never), CompilerGenerated> _
        Private __MechChecks As List(Of PrinterMechChecks)
        <CompilerGenerated, DebuggerBrowsable(DebuggerBrowsableState.Never)> _
        Private __PST As PST
    End Class
End Namespace

File: PST/FUEL Source Code/FUEL/TestSpecs.cs
// Decompiled with JetBrains decompiler
// Type: FUEL.TestSpecs
// Assembly: FUEL, Version=2.4.8.0, Culture=neutral, PublicKeyToken=null
// MVID: A3D630D0-0620-425D-8CFB-B67B2AC75507
// Assembly location: C:\hp\FUEL\FUEL.dll

using Microsoft.VisualBasic.CompilerServices;
using System.Collections.Generic;
using System.Diagnostics;

namespace FUEL
{
  public class TestSpecs
  {
    private string _TestID { [DebuggerNonUserCode] get; [DebuggerNonUserCode] set; }

    private List<PST.PrinterMechChecks> _MechChecks { [DebuggerNonUserCode] get; [DebuggerNonUserCode] set; }

    private PST _PST { [DebuggerNonUserCode] get; [DebuggerNonUserCode] set; }

    public string TestID
    {
      get => Operators.CompareString(this.PST.TestID, (string) null, false) != 0 ? this.PST.TestID : this._TestID;
      set => this._TestID = value;
    }

    public PST PST
    {
      get => this._PST;
      set => this._PST = value;
    }

    public TestSpecs() => this._MechChecks = new List<PST.PrinterMechChecks>();
  }
}
